packages/api/src/repository/library_item.ts

```diff
@@ -1,9 +1,26 @@
+import { DeepPartial } from 'typeorm'
 import { appDataSource } from '../data_source'
 import { LibraryItem } from '../entity/library_item'
+import { wordsCount } from '../utils/helpers'
 
 export const libraryItemRepository = appDataSource
   .getRepository(LibraryItem)
   .extend({
+    save(libraryItem: DeepPartial<LibraryItem>) {
+      return this.createQueryBuilder()
+        .insert()
+        .values({
+          ...libraryItem,
+          wordCount:
+            libraryItem.wordCount ??
+            wordsCount(libraryItem.readableContent || ''),
+        })
+        .orIgnore() // ignore if the item already exists
```

Comment:
If there was just a random failure, like a connection timeout would this be ignored?

Reply:
no, only unique constraint error will be ignored. i know this `orIgnore()` is misleading

Reply:
here the `orIgnore` is actually `on conflict do nothing`